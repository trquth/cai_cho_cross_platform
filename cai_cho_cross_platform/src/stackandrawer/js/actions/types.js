// league
export const GET_LEAGUE = 'GET_LEAGUE';

// match
export const GET_TEAM = 'GET_TEAM';

// navigation
export const TOGGLE_DRAWER = 'TOGGLE_DRAWER';
export const BACK = 'BACK';
