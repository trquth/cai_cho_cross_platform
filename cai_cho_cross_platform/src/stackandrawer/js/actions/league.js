import { GET_LEAGUE } from './types';
import { getTeams } from '../db/helpers';

export function getLeague(currentLeagueRoute) {
  const teams = getTeams(currentLeagueRoute);

  return {
    type: GET_LEAGUE,
    payload: { teams },
  };
}
