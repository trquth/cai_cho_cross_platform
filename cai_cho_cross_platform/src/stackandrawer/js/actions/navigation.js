import { BACK, TOGGLE_DRAWER } from './types';

export function back() {
  return { type: BACK };
}

export function toggleDrawer() {
  return { type: TOGGLE_DRAWER };
}
