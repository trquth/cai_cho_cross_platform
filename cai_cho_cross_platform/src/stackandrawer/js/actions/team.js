import { GET_TEAM } from './types';
import { getMatches } from '../db/helpers';
import { getTeamLogo } from '../api';

function teamAction(currentLeague, currentTeam, image) {
  const matches = getMatches(currentLeague, currentTeam);
  return { type: GET_TEAM, payload: { matches, currentTeam, image } };
}

export function getTeam(currentLeague, currentTeam) {
  return dispatch => getTeamLogo(currentTeam)
    .then(response => response.json())
    .then(response => dispatch(teamAction(currentLeague, currentTeam, response.Image)))
    .catch(() => dispatch(teamAction(currentLeague, currentTeam, null)))
  ;
}
