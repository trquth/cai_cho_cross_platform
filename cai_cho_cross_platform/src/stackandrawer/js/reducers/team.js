import { GET_TEAM } from './../actions/types';

const initialState = {
  image: '',
  currentTeam: '',
  matches: [],
};

export default function (state = initialState, action) {
  if (action.type === GET_TEAM) {
    return { ...action.payload };
  }

  return state;
}
