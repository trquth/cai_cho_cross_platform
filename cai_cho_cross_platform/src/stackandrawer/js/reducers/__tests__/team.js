import mockStore from 'redux-mock-store';
import { getTeam } from '../../actions/team';
import team from '../team';

const store = mockStore();

beforeEach(() => {
  store.clearActions();
});

it('INITIAL STATE', () => {
  const action = { type: 'DUMMY' };
  expect(team(undefined, action)).toMatchSnapshot();
});

it('GET_TEAM (LaLiga, Barcelona)', () => {
  expect(team(undefined, getTeam('LaLiga', 'Barcelona'))).toMatchSnapshot();
});
