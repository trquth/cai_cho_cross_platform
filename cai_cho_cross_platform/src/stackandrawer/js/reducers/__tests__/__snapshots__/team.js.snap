// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET_TEAM (LaLiga, Barcelona) 1`] = `
Object {
  "currentTeam": "",
  "image": "",
  "matches": Array [],
}
`;

exports[`INITIAL STATE 1`] = `
Object {
  "currentTeam": "",
  "image": "",
  "matches": Array [],
}
`;
